app/services/admin/update_induction_period.rb

```diff
@@ -16,20 +16,20 @@ def initialize(author:, induction_period:, params:)
     # @return [true]
     # @raise [ActiveRecord::RecordInvalid, ActiveRecord::Rollback]
     def update_induction_period!
-      validate_can_update!
-
       previous_start_date = induction_period.started_on
+      previous_end_date = induction_period.finished_on
       induction_period.assign_attributes(params)
       modifications = induction_period.changes
 
-      ActiveRecord::Base.transaction do
-        success = [
-          induction_period.save!,
-          record_event(modifications),
-          notify_trs_of_start_date_change(previous_start_date)
-        ].all?
+      # Prevent setting dates to nil for induction periods with outcomes
+      if induction_period.outcome.present? && params.key?(:finished_on) && params[:finished_on].nil?
+        raise ActiveRecord::RecordInvalid.new(induction_period), "End date cannot be set to nil for induction periods with outcomes"
```

Comment:
I don't think this validation scenario is in the feature spec but it'd be good to add. If the user can read this, we should reword it (unless @PriyaNahar2 disagrees)
 
`End date cannot be removed once the induction period has an outcome`